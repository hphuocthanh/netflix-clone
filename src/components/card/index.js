import React, { useState, useContext, createContext } from "react";

import {
  Container,
  Group,
  Title,
  Subtitle,
  Text,
  Entities,
  Meta,
  Item,
  Image,
  Feature,
  Content,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Maturity,
} from "./styles/card";

const FeatureContext = createContext();

export default function Card({ children, ...restOfProps }) {
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState(false);

  return (
    <>
      <FeatureContext.Provider
        value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
      >
        <Container {...restOfProps}>{children}</Container>
      </FeatureContext.Provider>
    </>
  );
}

Card.Group = function CardGroup({ children, ...restOfProps }) {
  return <Group {...restOfProps}>{children}</Group>;
};
Card.Title = function CardTitle({ children, ...restOfProps }) {
  return <Title {...restOfProps}>{children}</Title>;
};
Card.Subtitle = function CardSubtitle({ children, ...restOfProps }) {
  return <Subtitle {...restOfProps}>{children}</Subtitle>;
};

Card.Text = function CardText({ children, ...restOfProps }) {
  return <Text {...restOfProps}>{children}</Text>;
};

Card.Entities = function CardEntities({ children, ...restOfProps }) {
  return <Entities {...restOfProps}>{children}</Entities>;
};

Card.Meta = function CardMeta({ children, ...restOfProps }) {
  return <Meta {...restOfProps}>{children}</Meta>;
};

Card.Item = function CardItem({ item, children, ...restOfProps }) {
  const { setShowFeature, setItemFeature } = useContext(FeatureContext);

  return (
    <Item
      onClick={() => {
        setItemFeature(item);
        setShowFeature(true);
      }}
      {...restOfProps}
    >
      {children}
    </Item>
  );
};

Card.Image = function CardImage({ ...restOfProps }) {
  return <Image {...restOfProps} />;
};

Card.Feature = function CardFeature({ children, category, ...restOfProps }) {
  const { showFeature, itemFeature, setShowFeature } =
    useContext(FeatureContext);

  return showFeature ? (
    <Feature
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
    >
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowFeature(false)}>
          <img src="/images/icons/close.png" alt="Close" />
        </FeatureClose>

        <Group margin="30px 0" flexDirection="row" alignItems="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity < 12 ? "PG" : itemFeature.maturity}
          </Maturity>
          <FeatureText fontWeight="bold">
            {itemFeature.genre.charAt(0).toUpperCase() +
              itemFeature.genre.slice(1)}
          </FeatureText>
        </Group>

        {children}
      </Content>
    </Feature>
  ) : null;
};
