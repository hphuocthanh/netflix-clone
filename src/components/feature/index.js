import React from "react"
import { Container, Title, Subtitle } from "./styles/feature"

export default function Feature({ children, ...restOfProps}) {
    return (
        <Container {...restOfProps}>{children}</Container>
    )
}


Feature.Title = function FeatureTitle({ children, ...resOfProps}) {
    return (
        <Title {...resOfProps}>{children}</Title>
    )
}

Feature.Subtitle = function FeatureSubtitle({ children, ...resOfProps}) {
    return (
        <Subtitle {...resOfProps}>{children}</Subtitle>
    )
}