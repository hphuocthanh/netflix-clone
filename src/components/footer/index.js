import React from "react";
import {
  Container,
  Row,
  Column,
  Title,
  Link,
  Text,
  Break,
} from "./styles/footer";

export default function Footer({ children, ...restOfProps }) {
  return <Container {...restOfProps}>{children}</Container>;
}

Footer.Container = function FooterContainer({ children, ...restOfProps }) {};

Footer.Row = function FooterRow({ children, ...restOfProps }) {
  return <Row {...restOfProps}>{children}</Row>;
};

Footer.Column = function FooterColumn({ children, ...restOfProps }) {
  return <Column {...restOfProps}>{children}</Column>;
};

Footer.Title = function FooterTitle({ children, ...restOfProps }) {
  return <Title {...restOfProps}>{children}</Title>;
};

Footer.Link = function FooterLink({ children, ...restOfProps }) {
  return <Link {...restOfProps}>{children}</Link>;
};

Footer.Text = function FooterText({ children, ...restOfProps }) {
  return <Text {...restOfProps}>{children}</Text>;
};

Footer.Break = function FooterBreak({ children, ...restOfProps }) {
  return <Break {...restOfProps}>{children}</Break>;
};
