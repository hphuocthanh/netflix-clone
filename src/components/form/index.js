import React from "react";
import {
  Container,
  Input,
  Error,
  Submit,
  Base,
  Title,
  Text,
  TextSmall,
  Link,
  Provider,
  Icon
} from "./styles/form";

export default function Form({ children, ...restOfProps }) {
  return <Container {...restOfProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restOfProps }) {
  return <Base {...restOfProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restOfProps }) {
  return <Title {...restOfProps}>{children}</Title>;
};

Form.Text = function FormText({ children, ...restOfProps }) {
  return <Text {...restOfProps}>{children}</Text>;
};

Form.TextSmall = function FormTextSmall({ children, ...restOfProps }) {
  return <TextSmall {...restOfProps}>{children}</TextSmall>;
};

Form.Link = function FormLink({ children, ...restOfProps }) {
  return <Link {...restOfProps}>{children}</Link>;
};

Form.Input = function FormInput({ children, ...restOfProps }) {
  return <Input {...restOfProps}>{children}</Input>;
};

Form.Error = function FormError({ children, ...restOfProps }) {
  return <Error {...restOfProps}>{children}</Error>;
};

Form.Submit = function FormSubmit({ children, ...restOfProps }) {
  return <Submit {...restOfProps}>{children}</Submit>;
};

Form.Provider = function FormProvider({ children, ...restOfProps }) {
  return <Provider {...restOfProps}>{children}</Provider>;
};

Form.Icon = function FormIcon({ src, ...restOfProps }) {
  return <Icon src={src} {...restOfProps} />
};
