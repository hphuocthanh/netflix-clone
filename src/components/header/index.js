import React, { useState } from "react";
import { Link as RouterLink } from "react-router-dom";
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Group,
  Link,
  Text,
  Feature,
  FeatureCallOut,
  PlayButton,
  Search,
  SearchIcon,
  SearchInput,
  Picture,
  Profile,
  Dropdown,
} from "./styles/header";

export default function Header({
  isBgIncluded = true,
  children,
  ...restOfProps
}) {
  return isBgIncluded ? (
    <Background {...restOfProps}>{children}</Background>
  ) : (
    children
  );
}

Header.Frame = function HeaderFrame({ children, ...restOfProps }) {
  return <Container {...restOfProps}>{children}</Container>;
};

Header.Group = function HeaderGroup({ children, ...restOfProps }) {
  return <Group {...restOfProps}>{children}</Group>;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restOfProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restOfProps}>
      <SearchIcon onClick={() => setSearchActive(!searchActive)}>
        <img src="/images/icons/search.png" alt="Search " />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series"
        active={searchActive}
      />
    </Search>
  );
};

Header.Logo = function HeaderLogo({ to, ...restOfProps }) {
  return (
    <RouterLink to={to}>
      <Logo {...restOfProps} />
    </RouterLink>
  );
};

Header.Text = function HeaderText({ children, ...restOfProps }) {
  return <Text {...restOfProps}>{children}</Text>;
};

Header.Link = function HeaderLink({ children, ...restOfProps }) {
  return <Link {...restOfProps}>{children}</Link>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restOfProps }) {
  return <ButtonLink {...restOfProps}>{children}</ButtonLink>;
};

Header.Feature = function HeaderFeature({ children, ...restOfProps }) {
  return <Feature {...restOfProps}>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restOfProps
}) {
  return <FeatureCallOut {...restOfProps}>{children}</FeatureCallOut>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restOfProps }) {
  return <PlayButton {...restOfProps}>{children}</PlayButton>;
};

Header.Picture = function HeaderPicture({ src, ...restOfProps }) {
  return <Picture src={`/images/users/${src}.png`} {...restOfProps} />;
};

Header.Profile = function HeaderProfile({ children, ...restOfProps }) {
  return <Profile {...restOfProps}>{children}</Profile>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restOfProps }) {
  return <Dropdown {...restOfProps}>{children}</Dropdown>;
};
