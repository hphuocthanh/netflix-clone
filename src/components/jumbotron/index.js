import React from "react";
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  Subtitle,
  Image,
  Animation,
  AnimationText,
} from "./styles/jumbotron";

export default function Jumbotron({
  direction = "row",
  children,
  ...restOfProps
}) {
  return (
    <Item {...restOfProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
}

Jumbotron.Container = function JumbotronContainer({
  children,
  ...restOfProps
}) {
  return <Container {...restOfProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restOfProps }) {
  return <Pane {...restOfProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restOfProps }) {
  return <Title {...restOfProps}>{children}</Title>;
};

Jumbotron.Subtitle = function JumbotronSubtitle({ children, ...restOfProps }) {
  return <Subtitle {...restOfProps}>{children}</Subtitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restOfProps }) {
  return <Image {...restOfProps} />;
};

Jumbotron.Animation = function JumbotronAnimation({
  video,
  image2,
  text,
  text2,
  ...restOfProps
}) {
  return typeof video != "undefined" ? (
    <Animation {...restOfProps} watchOnTV>
      <video id="notflix-home-tvshow" playsInline autoPlay muted loop>
        <source src={video} type="video/mp4" />
      </video>
    </Animation>
  ) : (
    <Animation {...restOfProps} downloadAndWatch>
      <img src={image2} alt="" />
      <AnimationText>
        <div>{text}</div>
        <div>{text2}</div>
      </AnimationText>
    </Animation>
  );
};
