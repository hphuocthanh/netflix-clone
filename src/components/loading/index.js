import React from "react"
import { ReleaseBody, Spinner, LockBody, Picture } from "./styles/loading"

export default function Loading({ src, ...restOfProps }) {
    return (
        <Spinner {...restOfProps}>
            <LockBody />
            <Picture src={`/images/users/${src}.png`} />
        </Spinner>
    )
}

Loading.ReleaseBody = function LoadingReleaseBody() {
    return <ReleaseBody />
}
