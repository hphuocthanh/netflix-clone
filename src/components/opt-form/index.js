import React from "react";
import { Container, Input, Text, Button, Break } from "./styles/opt-form";

export default function OptForm({ children, ...restOfProps }) {
  return <Container {...restOfProps}>{children}</Container>;
}

OptForm.Input = function OptFormInput({ ...restOfProps }) {
  return <Input {...restOfProps} />;
};

OptForm.Text = function OptFormText({ children, ...restOfProps }) {
  return <Text {...restOfProps}>{children}</Text>;
};

OptForm.Button = function OptFormButton({ children, ...restOfProps }) {
  return (
    <Button {...restOfProps}>
      {children} <img src="/images/icons/chevron-right.png" alt="Try Now" />
    </Button>
  );
};

OptForm.Break = function OptFormBreak({ children, ...restOfProps }) {
    return <Break {...restOfProps}>{children}</Break>;
  };