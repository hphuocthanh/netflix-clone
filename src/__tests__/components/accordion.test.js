import React from "react";
import { render, fireEvent } from "@testing-library/react";
import faqsData from "../../fixtures/faqs.json";
import { Accordion } from "../../components";

describe("<Accordion />", () => {
  it("render the <Accordion /> with populated data", () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions?</Accordion.Title>
        <Accordion.Frame>
          {faqsData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Frame>
      </Accordion>
    );

    expect(getByText("Frequently Asked Questions?")).toBeTruthy();
    expect(getByText("What is Notflix?")).toBeTruthy();
    expect(getByText("How much does Notflix cost?")).toBeTruthy();
    expect(getByText("Where can I watch?")).toBeTruthy();
    expect(getByText("How do I cancel?")).toBeTruthy();
    expect(getByText("What can I watch on Notflix?")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("opens and closes the <Accordion /> component", () => {
    const { container, queryByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions?</Accordion.Title>
        <Accordion.Frame>
          {faqsData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Frame>
      </Accordion>
    );

    const whatIsNotflixText =
      "Notflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices.\n\nYou can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

    expect(queryByText(whatIsNotflixText)).toBeFalsy();
    fireEvent.click(queryByText("What is Notflix?"));
    expect(queryByText(whatIsNotflixText)).toBeTruthy();

    fireEvent.click(queryByText("What is Notflix?"))
    expect(queryByText(whatIsNotflixText)).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot()
  });
});
