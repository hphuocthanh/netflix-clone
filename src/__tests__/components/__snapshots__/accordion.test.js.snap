// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> render the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sc-1wnyt6p-0 ksMOZP"
>
  <div
    class="accordion__Inner-sc-1wnyt6p-2 jhrhpT"
  >
    <h1
      class="accordion__Title-sc-1wnyt6p-4 liYPYt"
    >
      Frequently Asked Questions?
    </h1>
    <div
      class="accordion__Frame-sc-1wnyt6p-1 bfUyVu"
    >
      <section
        class="accordion__Item-sc-1wnyt6p-3 ebdSZB"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 gbSJEf"
        >
          What is Notflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </section>
      <section
        class="accordion__Item-sc-1wnyt6p-3 ebdSZB"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 gbSJEf"
        >
          How much does Notflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </section>
      <section
        class="accordion__Item-sc-1wnyt6p-3 ebdSZB"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 gbSJEf"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </section>
      <section
        class="accordion__Item-sc-1wnyt6p-3 ebdSZB"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 gbSJEf"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </section>
      <section
        class="accordion__Item-sc-1wnyt6p-3 ebdSZB"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 gbSJEf"
        >
          What can I watch on Notflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </section>
    </div>
  </div>
</div>
`;
